src/devices/src/virtio/balloon/device.rs

```diff
@@ -0,0 +1,920 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cmp;
+use std::io::Write;
+use std::result;
+use std::result::Result;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use timerfd::{ClockId, SetTimeFlags, TimerFd, TimerState};
+
+use ::utils::eventfd::EventFd;
+use logger::{error, Metric, METRICS};
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use virtio_gen::virtio_blk::*;
+use vm_memory::{Address, ByteValued, Bytes, GuestAddress, GuestMemoryMmap};
+
+use super::*;
+use super::{
+    super::{
+        ActivateResult, DeviceState, Queue, VirtioDevice, TYPE_BALLOON, VIRTIO_MMIO_INT_VRING,
+    },
+    utils::{compact_page_frame_numbers, remove_range},
+};
+
+use crate::virtio::balloon::Error;
+use crate::{report_balloon_event_fail, Error as DeviceError};
+
+const SIZE_OF_U32: usize = 4;
```

Comment:
We can replace these constants with `std::mem::size_of::<u32>()` and `std::mem::size_of::<BalloonStat>()`. We can also add unit tests that assert the size of of BalloonsStat is 10.

Reply:
Done.